trifle/program/src/instruction.rs

```diff
@@ -150,6 +157,13 @@ pub enum TrifleInstruction {
     #[account(3, name = "system_program", desc = "System program")]
     AddTokensConstraintToEscrowConstraintModel(AddTokensConstraintToEscrowConstraintModelArgs),
 
+    #[default_optional_accounts]
+    #[account(0, writable, name = "constraint_model", desc = "Constraint model account")]
+    #[account(1, writable, signer, name = "payer", desc = "Wallet paying for the transaction")]
+    #[account(2, optional, signer, name = "update_authority", desc = "Update authority of the constraint model")]
```

Comment:
Why is this account optional in the JS given the on-chain handler expects it and explicitly checks it?

Reply:
This is in response to Keith's feedback but I think it actually should be explicitly required.